src/util/address.rs

```diff
@@ -287,34 +274,46 @@ impl Address {
             .push_int(0)
             .push_slice(&WPubkeyHash::from_engine(hash_engine)[..]);
 
+        let mut hash_engine = ScriptHash::engine();
+        hash_engine.input(&builder.into_script()[..]);
+
         Address {
             network: network,
-            payload: Payload::ScriptHash(ScriptHash::hash(builder.into_script().as_bytes())),
```

Comment:
Idem here. The builder has the byte slice internally, so `into_script()` just returns that slice as a script and `as_bytes` looks into the slice. So no allocation is prevented by using the engine here either.